test/cocotb/osdtestlib/soc_interface.py

```diff
@@ -0,0 +1,36 @@
+import cocotb
+from cocotb.triggers import Timer
+from cocotb.clock import Clock
+from cocotb.result import TestFailure
+from cocotb.triggers import RisingEdge
+
+from osdtestlib.debug_interconnect import RegAccess
+
+
+class TraceGenerator:
```

Comment:
This class is not generating generic traces, but only STM traces. Please rename accordingly.